plugins/sap-concur-approve-expense-report/README.md

```diff
@@ -17,7 +17,689 @@ systems:
+      "Resources": [
+          {
+              "displayName": "Kyle Hirai",
+              "id": "ad0dd555-2c34-4129-9d87-f7962b37cf56",
+              "userName": "kyle@moveworks.ai"
+          },
+          {
+              "displayName": "William Never",
+              "id": "ca5c42ed-06cb-431c-888a-369294210200",
+              "userName": "concur_admin@moveworks.ai"
+          },
+          {
+              "displayName": "ConcurConsultant ConcurConsultant",
+              "id": "9c10ad63-2774-476c-8ae2-0e2c18e7e27c",
+              "userName": "ConcurConsultant@p0030517jrwi"
+          },
+          {
+              "displayName": "Concur Administrator",
+              "id": "80f30214-1010-48ba-ab3e-374559f32a6c",
+              "userName": "ConcurAdmin@p0030517jrwi"
+          },
+          ...
+      ]
+  }
+  ```
+
+### **API 3: Get Reports to Approve by User**
+
+```bash
+curl --location --request GET 'https://us2.api.concursolutions.com/expensereports/v4/users/{user_id}/context/MANAGER/reportsToApprove?sort=reportDate&order=desc&includeDelegateApprovals=true' \
+--header 'Accept: application/json' \
+--header 'Authorization: Bearer {auth_token}' \
+```
+
+### **API 4: Moveworks Events API**
+
+Please replace the `email@example.com` with the email address of the user who will receive the notification.
+
+You can get the `{{moveworks_event_id}}` and `{{moveworks_events_api_key}}` from following the quickstart guide [here](https://developer.moveworks.com/creator-studio/quickstart/event-triggered-paths/).
+
+```bash
+curl --location 'https://api.moveworks.ai/rest/v1/events/{{moveworks_event_id}}/messages/send' \
+--header 'Authorization: Bearer {{moveworks_events_api_key}}' \
+--header 'Content-Type: application/json' \
+--data-raw '{
+    "message": "💸 New Concur Expense Report Approval Request ....\n",
+    "recipients": [
+        "email@example.com"
+    ],
+    "context": {
+        "slots": {
+            "report_id": "692D37074CAA42C2BE47"
+        }
+    }
+}'
+```
+
+### **API 5: Approve Expense Report (as Approver)**
+
+```bash
+curl --location --request PATCH 'https://us2.api.concursolutions.com/expensereports/v4/users/{user_id}/context/MANAGER/reports/{report_id}/approve' \
+--header 'Accept: application/json' \
+--header 'Authorization: Bearer {auth_token}' \
+--data '{
+    "comment":"Approved through API"
+}'
+```
+
+### **API 6: Send Back Expense Report (as Approver)**
+
+```bash
+curl --location --request PATCH 'https://us.api.concursolutions.com/expensereports/v4/users/{user_id}/context/MANAGER/reports/{report_id}/sendBack' \
+--header 'Authorization: Bearer {auth_token}' \
+--header 'Content-Type: application/json' \
+--data '{
+    "comment":"Rejecting",
+    "expenseRejectedComment": "Rejecting Report through API"
+}'
+```
+
+## Step 3: Write your Orchestration Code in your Middleware Platform
+
+This diagram describes the flow we will build in this guide.
+
+
+  ![image.png](./architecture.png)
```

Comment:
![CleanShot 2024-10-31 at 08 17 57@2x](https://github.com/user-attachments/assets/e657ebd9-325f-4f12-9b23-76d81bb12138)

Reply:
We can also drop the user identity one